### DISCUSSION TITLE
The running time of the 'canneal' benchmark is quite unstable.

### DISCUSSION THREAD

**Discussion by hisengberg** [anujpathania/HotSniper#47] [Q&A]
Dear Professor,
When I run the _canneal_ benchmark repeatedly **under the same configuration**, no matter what mapping policy or dvfs policy is used, the result of final runtime is unstable, there will be a relatively large difference in the final runtime. However, when I ran other benchmarks, under the same configuration, the running time was relatively stable, and there was no such situation as _canneal_. 
I want to know why this happens? I searched for relevant information and learned that _canneal_ is strongly related to main memory, and the _**Sniper**_ in **_HotSniper_** does not seem to model the main memory capacity. Is this related to it?
![image](https://github.com/anujpathania/HotSniper/assets/118739669/8aa01df5-d91c-4fc7-b6ce-ba72d4a2f6f0)

**Comment by anujpathania** [comment]
Hi,

Use of Pin in Sniper inherently brings in some randomness in the final result. It is also affected by OS activity to some extent, which adds to the entropy.

Why the entropy is higher for Cannel than other benchmarks, I do not know. Cannel nevertheless is a highly memory-intensive benchmark.

Sniper does model main memory capacity. By default it is 1024 MB I think, and you can change it in base config. 

Sniper doesn't have concept of virtual memory, so I don't know what will happen if this main memory gets exhausted. Since we run mostly sim-small and sim-medium benchmarks in HotSniper .. 1 GB is more than enough.

-- Anuj

**Reply by hisengberg** [reply]
Dear Anuj,

I am greatly appreciative for your prompt response.

I am slightly perplexed by your statement "_Sniper does model main memory capacity. By default it is 1024 MB I think, and you can change it in base config_." as I was unable to locate the corresponding settings within "`HotSniper/configbase.cfg`".

Furthermore, I came across a similar post within the _Sniper_ forum where the Sniper developers explicitly stated that they did not model the main memory capacity. and I've attached the link below.
![image](https://github.com/anujpathania/HotSniper/assets/118739669/7a8e13ff-6316-4d4a-913d-bd369fc0b3f1)


 **But I’m not sure if Sniper has subsequently updated the relevant settings.** 

Regarding the default value you mentioned, which is **1024MB**, I did discover the corresponding value within the file of "`HotSniper/config/dram-cache.cfg`". Is this what you are talking about?
![image](https://github.com/anujpathania/HotSniper/assets/118739669/d88e77c3-bfa4-4281-9495-9c09b13632c9)


[https://groups.google.com/g/snipersim/c/uaThXL7-jo4](url)

**Reply by anujpathania** [reply]
Yes. This is exactly the parameter I was referring to.

What else it could be other than the size of the main memory.

-- Anuj
